packages/engine-core/src/data-proxy/utils/getClientVersion.ts

```diff
@@ -1,19 +1,42 @@
+import Debug from '@prisma/debug'
+
 import type { EngineConfig } from '../../common/Engine'
+import { NotImplementedYetError } from '../errors/NotImplementedYetError'
+
+const debug = Debug('prisma:engine-core:getClientVersion')
 
 /**
  * Determine the client version to be sent to the DataProxy
  * @param config
  * @returns
  */
-export function getClientVersion(config: EngineConfig) {
-  const [version, suffix] = config.clientVersion?.split('-') ?? []
+export function getClientVersion(config: EngineConfig): string {
+  const dataproxyClientVersionFromEnvVar = process.env.PRISMA_CLIENT_DATA_PROXY_CLIENT_VERSION
+
+  if (dataproxyClientVersionFromEnvVar) {
+    debug(`Client version: "${dataproxyClientVersionFromEnvVar}" from PRISMA_CLIENT_DATA_PROXY_CLIENT_VERSION env var.`)
+    return dataproxyClientVersionFromEnvVar
+  } else if (config.clientVersion) {
+    const [version, suffix] = config.clientVersion.split('-') ?? []
```

Comment:
I believe `?? []` is a no-op here, `String.prototype.split` always returns an array

```suggestion
    const [version, suffix] = config.clientVersion.split('-')
```